src/Lib/Buddy.php

```diff
@@ -0,0 +1,245 @@
+		// so far only use case with non-existing local index
+		if (preg_match('/index (.*?) absent/', $this->origMsg)) {
+			return 'NO_INDEX';
+		}
+		if (strpos($this->origMsg, 'unexpected identifier') !== false) {
+			return 'UNKNOWN_COMMAND';
+		}
+		return '';
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function detectQueryType(): string {
+		if (stripos($this->query, 'INSERT') === 0) {
+			return 'INSERT_QUERY';
+		}
+		if (stripos($this->query, 'SHOW QUERIES') === 0) {
+			return 'SHOW_QUERIES_QUERY';
+		}
+		return '';
+	}
+
+	/**
+	 * @param ACTION $action
+	 * @return string
+	 */
+	protected function buildMntRequest(ACTION $action): string {
+		switch ($action) {
+			case ACTION::INSERT:
+				return $this->query;
+			case ACTION::CREATE_INDEX:
+				return $this->buildMntRequestWithHelpers('INSERT', 'CREATE');
+			case ACTION::SELECT_SYSTEM_SESSIONS:
+				return 'SELECT connid AS ID, `last cmd` AS query, host, proto FROM @@system.sessions)';
+			default:
+				return '';
+		}
+	}
+
+	/**
+	 * @param string $parseStmt
+	 * @param string $buildStmt
+	 * @return string
+	 * @throws Throwable
+	 */
+	protected function buildMntRequestWithHelpers(string $parseStmt, string $buildStmt): string {
+		try {
+			if (!method_exists($this->queryParserLocator, 'getQueryParser')) {
+				$this->error('Required parser cannot be found');
+			} else {
+				$parser = $this->queryParserLocator->getQueryParser($this->reqFormat, $parseStmt);
+				if (method_exists($parser, 'parse')) {
+					$queryData = $parser->parse($this->query);
+					if (array_key_exists('error', $queryData)) {
+						return $queryData['error'];
+					}
+					if (!method_exists($this->stmtBuilder, 'build')) {
+						$this->error('Statement builder cannot be found');
+					}
+					return $this->stmtBuilder->build($buildStmt, $queryData);
+				}
+				$this->error('Unvalid parser detected');
+			}
+		} catch (Throwable $e) {
+			$this->error($e->getMessage());
+		}
+		return '';
+	}
+
+	/**
+	 * @param string $mntRequest
+	 * @return string
+	 */
+	protected function getMntResponse(string $mntRequest): string {
+		$resp  = '';
+		$respEndpoint = "http://{$this->clAddrPort}/cli";
+		$conn = curl_init($respEndpoint);
+		if ($conn === false) {
+			$this->error("Cannot conect to server at $respEndpoint");
+		} else {
+			curl_setopt($conn, CURLOPT_RETURNTRANSFER, true);
+			curl_setopt($conn, CURLOPT_POST, 1);
+			curl_setopt($conn, CURLOPT_POSTFIELDS, $mntRequest);
+			curl_setopt($conn, CURLOPT_HTTPHEADER, ['Content-Type: text/plain']);
+			curl_setopt($conn, CURLOPT_HTTPHEADER, ['X-Manticore-Error-Redirect: disable']);
+			$resp = (string)curl_exec($conn);
+			curl_close($conn);
+		}
+		return $resp;
+	}
+
+	/**
+	 * @return bool
+	 */
+	protected function isMntResponseOk(): bool {
+		$resp = json_decode($this->msg, true);
+		if ($resp === null) {
+			return false;
+		}
+		$resp = (array)$resp;
+		return (array_key_exists('error', $resp) && $resp['error'] !== '') ? false : true;
```

Comment:
Maybe, `return (bool)($resp['error'] ?? true);`
If an error exists and is not an empty string? Maybe it's better to set it to exist ONLY when there is an error OR exist all the time but with no value when no error.